lib/logger.js

```diff
@@ -222,22 +222,21 @@ Logger.prototype._bufferLog = function(message) {
 
     if (!this._flusher) {
         debug('No scheduled flush. Scheduling for %d ms from now.', configs.FLUSH_INTERVAL);
-        this._flusher = setTimeout(bind(this._flush, this, (err) => {
-            if (err) {
-                debug('Received an error while flushing...' + err);
+        this._flusher = setTimeout(() => {
+            if (this._buf.length > 0) {
```

Comment:
this conditional check here is going to be problematic, because if `this._buf.length` is 0, then `_flush` isn't called and then the flushing mechanism just stops without an ability to start again.